services/server/src/server/services/utils/contract-creation-util.ts

```diff
@@ -1,74 +1,285 @@
+    apiURL + BLOCKSSCAN_SUFFIX,
+    (response: any) => {
+      if (response.fromTxn) return response.fromTxn as string;
     }
+  );
+}
 
-    // XDC
-    if (sourcifyChain.chainId == 50 || sourcifyChain.chainId == 51) {
-      const response = await fetchFromApi(contractFetchAddressFilled);
-      if (response.fromTxn) return response.fromTxn as string;
+function getMeterApiContractCreatorFetcher(
+  apiURL: string
+): ContractCreationFetcher {
+  return getApiContractCreationFetcher(
+    apiURL + METER_SUFFIX,
+    (response: any) => {
+      return response.account.creationTxHash as string;
     }
+  );
+}
 
-    // Meter network
-    if (sourcifyChain.chainId == 83 || sourcifyChain.chainId == 82) {
-      const response = await fetchFromApi(contractFetchAddressFilled);
-      if (response?.account?.creationTxHash)
-        return response.account.creationTxHash as string;
+function getTelosApiContractCreatorFetcher(
+  apiURL: string
+): ContractCreationFetcher {
+  return getApiContractCreationFetcher(
+    apiURL + TELOS_SUFFIX,
+    (response: any) => {
+      if (response.creation_trx) return response.creation_trx as string;
     }
+  );
+}
 
-    // Avalanche Subnets
-    if (contractFetchAddressFilled.includes("avax.network")) {
-      const response = await fetchFromApi(contractFetchAddressFilled);
+function getAvalancheApiContractCreatorFetcher(
+  chainId: string
+): ContractCreationFetcher {
+  return getApiContractCreationFetcher(
+    `https://glacier-api.avax.network/v1/chains/${chainId}/${AVALANCHE_SUBNET_SUFFIX}`,
+    (response: any) => {
       if (response.nativeTransaction?.txHash)
         return response.nativeTransaction.txHash as string;
     }
+  );
+}
+
+async function getCreatorTxUsingFetcher(
+  fetcher: ContractCreationFetcher,
+  contractAddress: string
+) {
+  if (fetcher === undefined) {
+    return null;
+  }
+
+  const contractFetchAddressFilled = fetcher?.url.replace(
+    "${ADDRESS}",
+    contractAddress
+  );
+
+  if (!contractFetchAddressFilled) return null;
+
+  try {
+    switch (fetcher.type) {
+      case "scrape": {
+        if (fetcher?.scrapeRegex) {
+          const creatorTx = await getCreatorTxByScraping(
+            contractFetchAddressFilled,
+            fetcher?.scrapeRegex
+          );
+          if (creatorTx) return creatorTx;
+        }
+        break;
+      }
+      case "api": {
+        if (fetcher?.responseParser) {
+          const response = await fetchFromApi(contractFetchAddressFilled);
+          return fetcher?.responseParser(response);
+        }
+        break;
+      }
+    }
   } catch (e: any) {
     logger.warn("Error while getting creation transaction: " + e.message);
     return null;
   }
 
+  return null;
+}
+
+/**
+ * Finds the transaction that created the contract by either scraping a block explorer or querying a provided API.
+ *
+ * @param sourcifyChain
+ * @param address
+ * @returns
+ */
+export const getCreatorTx = async (
```

Comment:
I think in this function we should prioritize the APIs, and move the scrape methods to the bottom